packages/fether-react/src/App/App.js

```diff
@@ -42,6 +45,22 @@ class App extends Component {
   };
 
   componentDidMount () {
+    if (!electron) {
+      return;
+    }
+
+    if (store.get(LANG_LS_KEY) && i18n.language !== store.get(LANG_LS_KEY)) {
+      i18n.changeLanguage(store.get(LANG_LS_KEY));
+    }
+
+    electron.remote
+      .getCurrentWindow()
+      .webContents.addListener('set-language', newLanguage => {
+        i18n.changeLanguage(newLanguage);
+        store.set(LANG_LS_KEY, newLanguage);
```

Comment:
If I understood correctly, this setting is already saved in `electron-settings`. Isn't saving it in localStorage redundant?

Reply:
When I was deciding where to save the user's language setting I found I only had access to electron-settings in the fether-electron package, but not in fether-react (and I don't think we'd want the front-end to have access to the file system anyway). I need to store the language in localStorage too since I'm using i18n in the front-end for the Fether contents, and when it initialises it sets the language to `lng` in /Users/scon/code/src/paritytech/fether/packages/fether-electron/src/main/app/menu/i18n/index.js based on the language last saved in localStorage or English as a fallback, and when the App.js component finishes mounting it checks if the language saved in localStorage differs from the currently set i18n language, and if so it retrieves the latest localStorage language value and sets that to the i18n language (see /Users/scon/code/src/paritytech/fether/packages/fether-react/src/App/App.js).  

**fether-electron** has the Fether menu language config.
* i18n config initialises to the language stored in electron-settings under key `fether-language`, or uses fallback default language of English (see fether-electron/src/main/app/menu/i18n/index.js)
* When it's setting up the Fether menu it determines which language checkbox to tick based on the value of the key `fether-language` stored in electron-settings. See /Users/scon/code/src/paritytech/fether/packages/fether-electron/src/main/app/menu/template/index.js
* If the user clicks one of the languages in the Fether menu, then it triggers the click handler, which changes the fether-electron i18n language, sets the `fether-language` key in electron-settings to the language that was clicked, updates the Fether menu, and emits `set-language', which the fether-react event listener handles by changing the fether-react i18n language, sets the `fether-language` key in localStorage to the language that was clicked, and reloads the Fether window. See /Users/scon/code/src/paritytech/fether/packages/fether-react/src/App/App.js

**fether-react** has the i18n Fether contents language config
* i18n config initialises to the language stored in localStorage under key `fether-language`, or uses fallback default language of English (see /Users/scon/code/src/paritytech/fether/packages/fether-react/src/i18n/index.js)

Reply:
> i18n config initialises to the language stored in localStorage

My idea is:
- fether-electron stores user's language in `electron-settings` - we agree on that
- On startup (i.e. `componendDidMount` inside App.js), react reads from IPC from electron (i.e. using `electron.remote`) and sets the language accordingly
- Everytime react receives a `set-language` IPC message, it changes the language accordingly

There's really no need for localStorage

Reply:
But the problem is that after the user changes the language in the UI, first the `click ()` handler for the language they chose changes the language in fether-electron's i18n config and updates the Fether menu, which successfully changes the language shown in the Fether menu (but not the Fether window contents), but then it emits the `'set-language'` event which triggers the `addListener('set-language'` block in `componendDidMount` of App.js where it changes the i18n language in the front-end to match the new language stored in `electron-settings`. But then in order for the Fether window contents to change to the new language that was emitted, we need to reload the Fether window, which is why we end the block with `electron.remote.getCurrentWindow().webContents.reload();`, but when it reloads, it also reloads the front-end i18n configuration back to its defaults.

Reply:
I've created a separate branch where I've added two additional commits, one that merges the German language branch (so you can see how it works switching between languages), and a second commit where i've tried get your idea to work unsuccessfully https://github.com/paritytech/fether/compare/luke-402-i18n-temp?expand=1